File: tests/java/com/trulioo/sdk/model/DriverLicenceTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class DriverLicenceTest {
    private DriverLicence driverLicence;

    @BeforeEach
    public void beforeEach() {
        driverLicence = new DriverLicence();
    }

    @Test
    public void testNumber() {
        String number = "123";
        driverLicence.setNumber(number);

        assertEquals(number, driverLicence.getNumber());
        assertEquals(number, new DriverLicence().number(number).getNumber());
    }

    @Test
    public void testState() {
        String state = "Alberta";
        driverLicence.setState(state);

        assertEquals(state, driverLicence.getState());
        assertEquals(state, new DriverLicence().state(state).getState());
    }

    @Test
    public void testDayOfExpiry() {
        Integer dayOfExpiry = 15;
        driverLicence.setDayOfExpiry(dayOfExpiry);

        assertEquals(dayOfExpiry, driverLicence.getDayOfExpiry());
        assertEquals(dayOfExpiry, new DriverLicence().dayOfExpiry(dayOfExpiry).getDayOfExpiry());
    }

    @Test
    public void testMonthOfExpiry() {
        Integer monthOfExpiry = 9;
        driverLicence.setMonthOfExpiry(monthOfExpiry);

        assertEquals(monthOfExpiry, driverLicence.getMonthOfExpiry());
        assertEquals(monthOfExpiry, new DriverLicence().monthOfExpiry(monthOfExpiry).getMonthOfExpiry());
    }

    @Test
    public void testYearOfExpiry() {
        Integer yearOfExpiry = 2020;
        driverLicence.setYearOfExpiry(yearOfExpiry);

        assertEquals(yearOfExpiry, driverLicence.getYearOfExpiry());
        assertEquals(yearOfExpiry, new DriverLicence().yearOfExpiry(yearOfExpiry).getYearOfExpiry());
    }

    @Test
    public void testEquals() {
        String number = "123";
        DriverLicence driverLicence1 = new DriverLicence().number(number);

        assertEquals(driverLicence1, driverLicence1);
        assertEquals(driverLicence1, new DriverLicence().number(number));
        assertNotEquals(driverLicence1, new DriverLicence().number(number + "1"));
        assertNotEquals(driverLicence1, null);
    }

    @Test
    public void testHashcode() {
        String number = "123";

        assertEquals(new DriverLicence().number(number).hashCode(), new DriverLicence().number(number).hashCode());
    }

    @Test
    public void testToString() {
        String number = "123";

        assertEquals(new DriverLicence().number(number).toString(), new DriverLicence().number(number).toString());
    }
}

File: tests/php/Model/DriverLicenceTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\DriverLicence;

class DriverLicenceTest extends TestCase
{
    private DriverLicence $driverLicence;

    public function setUp(): void
    {
        $this->driverLicence = new DriverLicence([]);
    }

    public function testPropertyNumber()
    {
        $this->driverLicence->setNumber('123');
        $this->assertEquals('123', $this->driverLicence->getNumber());
    }

    public function testPropertyState()
    {
        $this->driverLicence->setState('BC');
        $this->assertEquals('BC', $this->driverLicence->getState());
    }

    public function testPropertyDayOfExpiry()
    {
        $this->driverLicence->setDayOfExpiry(15);
        $this->assertEquals(15, $this->driverLicence->getDayOfExpiry());
    }

    public function testPropertyMonthOfExpiry()
    {
        $this->driverLicence->setMonthOfExpiry(9);
        $this->assertEquals(9, $this->driverLicence->getMonthOfExpiry());
    }

    public function testPropertyYearOfExpiry()
    {
        $this->driverLicence->setYearOfExpiry(2020);
        $this->assertEquals(2020, $this->driverLicence->getYearOfExpiry());
    }
}
